src/transforms/kv_parser.rs

```diff
@@ -0,0 +1,404 @@
+use super::Transform;
+use crate::{
+    event::{self, Event},
+    topology::config::{DataType, TransformConfig, TransformContext, TransformDescription},
+    types::{parse_conversion_map, Conversion},
+};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::str;
+use string_cache::DefaultAtom as Atom;
+
+#[derive(Debug, Default, Derivative, Deserialize, Serialize)]
+#[serde(default, deny_unknown_fields)]
+pub struct KeyValueConfig {
+    #[derivative(Default(value = "true"))]
+    pub drop_field: bool,
+    pub field: Option<Atom>,
+    #[derivative(Default(value = " "))]
+    pub field_split: String,
+    #[derivative(Default(value = "true"))]
+    pub overwrite_target: bool,
+    pub separator: String,
+    pub target_field: Option<Atom>,
+    pub trim_key: Option<String>,
+    pub trim_value: Option<String>,
+    pub types: HashMap<Atom, String>,
+}
+
+inventory::submit! {
+    TransformDescription::new::<KeyValueConfig>("kv_parser")
+}
+
+#[typetag::serde(name = "kv_parser")]
+impl TransformConfig for KeyValueConfig {
+    fn build(&self, _cx: TransformContext) -> crate::Result<Box<dyn Transform>> {
+        let field = self
+            .field
+            .as_ref()
+            .unwrap_or(&event::log_schema().message_key());
+
+        let conversions = parse_conversion_map(&self.types)?;
+
+        Ok(Box::new(KeyValue {
+            conversions,
+            drop_field: self.drop_field,
+            field: field.clone(),
+            field_split: self.field_split.clone(),
+            overwrite_target: self.overwrite_target,
+            separator: self.separator.clone(),
+            target_field: self.target_field.clone(),
+            trim_key: self.trim_key.clone(),
+            trim_value: self.trim_value.clone(),
+        }))
+    }
+
+    fn input_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    fn output_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    fn transform_type(&self) -> &'static str {
+        "kv_parser"
+    }
+}
+
+pub struct KeyValue {
+    conversions: HashMap<Atom, Conversion>,
+    drop_field: bool,
+    field: Atom,
+    field_split: String,
```

Comment:
Given it's usage, this would probably be better as an `Option<String>` where `None` represents splitting on white space.